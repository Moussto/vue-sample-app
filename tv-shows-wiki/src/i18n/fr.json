{
  "app-title": "TV Shows Wiki",
  "tv-show-card": {
    "expand": "Voir plus",
    "created-in": "Créée en",
    "unknown-year" : "Date de création non renseignée",
    "is-favorite" : "Favorie !"
  },
  "show-detail": {
    "favorite-button": "Ajouter aux favoris",
    "remove-favorite-button": "Enlever des favoris"
  }
}
