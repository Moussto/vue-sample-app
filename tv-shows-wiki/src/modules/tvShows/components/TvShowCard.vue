<template>
  <md-card class="show md-elevation-4">
    <md-card-media @click="goToShowPage">
      <img class="show__poster" @click="goToShowPage" :src="tvShow.images.poster" alt="ShowImage">
    </md-card-media>

    <md-card-header>
      <div class="md-title show__title" @click="goToShowPage">{{ tvShow.title }}</div>
      <div class="md-subhead" v-if="tvShow.yearOfCreation">{{ createdInLabel }}</div>
      <div class="md-subhead" v-else>{{ $t("tv-show-card.unknown-year") }}</div>
    </md-card-header>

    <md-card-expand>
      <md-card-actions md-alignment="space-between">
        <md-card-expand-trigger>
          <md-button class="show__expand-button">{{ $t('tv-show-card.expand')}}</md-button>
        </md-card-expand-trigger>
        <md-badge v-if="tvShow.isFavorite" class="md-square favorite-badge" :md-content="$t('tv-show-card.is-favorite')" />
      </md-card-actions>

      <md-card-expand-content>
        <md-card-content>
          <span class="show__description">{{ tvShow.description }}</span>
        </md-card-content>
      </md-card-expand-content>
    </md-card-expand>
  </md-card>
</template>
<script>
import TvShow from '@/modules/tvShows/entity/tvShow.model'

export default {
  name: 'tv-show-card',
  computed: {
    createdInLabel () {
      return `${this.$t('tv-show-card.created-in')} ${this.tvShow.yearOfCreation}`
    }
  },
  methods: {
    goToShowPage () {
      this.$router.push({ name: 'profileShow', params: { showId: this.tvShow.id } })
    }
  },
  props: {
    tvShow: {
      type: TvShow,
      required: true
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
  .md-card {
    width: 290px;
    margin: 4px 0 25px;
    display: inline-block;
    vertical-align: top;
  }

  .show__title {
    cursor: pointer;
    &:hover {
      color: deepskyblue;
    }
  }

  .favorite-badge {
    margin-right: 20px;
  }

  .show__poster {
    cursor: pointer;
  }
</style>
