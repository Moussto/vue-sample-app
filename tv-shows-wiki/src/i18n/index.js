import Vue from 'vue'
import VueI18n from 'vue-i18n'

import fr from '@/i18n/fr.json'

Vue.use(VueI18n)

const messages = {
  fr
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'fr',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'fr',
  messages
})
