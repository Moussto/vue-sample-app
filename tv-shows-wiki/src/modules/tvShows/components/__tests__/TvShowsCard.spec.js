import Vue from 'vue'
import { mount } from '@vue/test-utils'
import i18n from '@/i18n'
import TvShowCard from '../TvShowCard.vue'
import TvShow from '@/modules/tvShows/entity/tvShow.model'

// we register VueMaterial here, otherwise the Vue instance in tests wouldn't recognize elements like <md-card/>
// we also could have stubbed those elements
import VueMaterial from 'vue-material'
Vue.use(VueMaterial)

describe('TvShowCard.vue', () => {
  let options = null
  let wrapper = null

  beforeEach(() => {
    options = {
      i18n,
      propsData: {
        tvShow: new TvShow({
          id: 1,
          title: 'OSB show',
          description: 'Description de la série OSB show...',
          yearOfCreation: '2015'
        })
      }
    }
  })

  it('should render correctly TvShowCard', () => {
    // GIVEN the mounted component
    wrapper = mount(TvShowCard, options)
    // THEN it should match snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should display error message if creation year is not given', () => {
    // GIVEN a tv show with no year of creation
    options.propsData.tvShow = new TvShow({ id: 1 })
    // WHEN mounting component
    wrapper = mount(TvShowCard, options)
    // THEN it should display message instead of year
    expect(wrapper.find('.md-subhead').text()).toEqual('Date de création non renseignée')
  })
  it('should push show page route when clicking on title', () => {
    // GIVEN a mocked router
    wrapper = mount(TvShowCard, {
      ...options,
      propsData: {
        tvShow: new TvShow({ id: 42 })
      },
      mocks: {
        $router: {
          push: jest.fn()
        }
      }
    })
    // WHEN clicking on title
    wrapper.find('.show__title').trigger('click')
    // THEN router should be called with profileShow page and correct showId
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'profileShow', params: { showId: 42 } })
  })
  it('should display favorite badge if show is one', () => {
    // GIVEN a tv show that is in favorites
    options.propsData.tvShow = new TvShow({ id: 1, isFavorite: true })
    // WHEN mounting component
    wrapper = mount(TvShowCard, options)
    // THEN it should display a favorite badge
    expect(wrapper.find('.favorite-badge').exists()).toBe(true)
  })
})
