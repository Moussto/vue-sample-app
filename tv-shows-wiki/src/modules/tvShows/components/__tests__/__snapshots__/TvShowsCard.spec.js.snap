// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowCard.vue should render correctly TvShowCard 1`] = `
<div class="md-card show md-elevation-4 md-theme-default">
  <div class="md-card-media"><img alt="ShowImage" class="show__poster"></div>
  <div class="md-card-header">
    <div class="md-title show__title">OSB show</div>
    <div class="md-subhead">Créée en 2015</div>
  </div>
  <div class="md-card-expand">
    <div class="md-card-actions md-alignment-space-between"><button type="button" class="md-button show__expand-button md-card-expand-trigger md-theme-default">
        <div class="md-ripple">
          <div class="md-button-content">Voir plus</div>
        </div>
      </button>
      <!---->
    </div>
    <div class="md-card-expand-content" style="margin-top: -0px; opacity: 1;">
      <div class="md-card-content"><span class="show__description">Description de la série OSB show...</span></div>
    </div>
  </div>
</div>
`;
